runtime/drivers/clickhouse/model_executor_s3_self.go

```diff
@@ -75,11 +98,19 @@ func (e *s3ToSelfExecutor) Execute(ctx context.Context, opts *drivers.ModelExecu
 		return nil, err
 	}
 
-	// Build the model executor options with updated input properties
+	// Build the model executor options with updated input and output properties
 	clone := *opts
 	clone.InputProperties = propsMap
 	newOpts := &clone
 
+	var outputPropsMap map[string]any
+	err = mapstructure.WeakDecode(outputProps, &outputPropsMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to set output properties: %w", err)
+	}
+	newOpts.OutputProperties = outputPropsMap
```

Comment:
```suggestion
	// Ensure materialize is true because the selfToSelfExecutor is not able to infer it independently.
	outputProps := &outputProperties{}
	err = mapstructure.WeakDecode(opts.OutputProperties, &outputProps)
	if err != nil {
		return nil, fmt.Errorf("failed to parse output properties: %w", err)
	}
	truth := true
	outputProps = &truth
	err = mapstructure.WeakDecode(outputProps, &newOpts.OutputProperties)
	if err != nil {
		return nil, fmt.Errorf("failed to parse output properties: %w", err)
	}
```

Reply:
Sorry this is unclear to me. 
Do you mean we handle the validation(that materialize should not be false) in the `validate` function and handle setting the default here ?

In that case I think we can avoid the `validate` function as well and directly throw error here. 
We can also avoid create a separate `outputProperties` struct and directly parse it `ModelOutputProperties` ?
This is what I have in my mind : https://github.com/rilldata/rill/pull/5422/commits/43b0af324158b72a53cc81980bfeff21e221507e

Reply:
Okay I need `outputProperties` for a separate change so we can keep it and its validation function.